### ISSUE TITLE
Allow filtering for complex types on GraphQL API

### ISSUE THREAD

**Issue by poornimanayar** [umbraco/Umbraco.Cloud.Issues#303]
Hi,

I have been playing around with the Heartcore GraphQL API and I feel one of the features it lacks is to allow filtering by complex types like MNTP, Content Picker, etc. The only workaround at the moment is to filter it out on the client-side which is not ideal at all. I am looking at integrating a Blazor SPA with Heartcore and for me, GraphQL means a totally fluid way of querying and retrieving data. While this might not pose a major issue with websites, instances, where Heartcore is purely a content hub for relational data and the client, is a 360-degree view of that data, this can pose a real problem for the developers.

Poornima

**Comment by dawoe** [comment]
Just running into this as well. Any updates on this ?

**Comment by dawoe** [comment]
Because of this I can't use the built in pagination.

**Comment by sitereactor** [comment]
@dawoe I'm not entirely sure I understand your comments here. Poornima's post is a feature request. Are you two comments about this issue that you posted yesterday https://github.com/umbraco/Umbraco.Cloud.Issues/issues/352 or something differently?

**Comment by dawoe** [comment]
Nope this missing feature is a issue for us as well now. 

I relate some nodes using MNTP in the backoffice. But filtering on a MNTP selection is not possible at the moment.

So we need to get all the data to our client app and do filtering and pagination there.

For simple datatypes (like textbox) we can do filtering and pagination in the graphQL query.

**Comment by sitereactor** [comment]
Is it possible to expand on this with a usecase or scenario for filtering items based on MNTP? Just trying to better understand what you want to achieve (aka the end goal).

**Comment by dawoe** [comment]
Yep... I need something similar as Poornima, we have a product that can be tagged with multiple categories and shops (using MNTP) that it is available in. I want to filter on category and shop.

**Comment by sitereactor** [comment]
Thanks for elaborating. It makes sense and I agree that it would be a nice addition. We'll keep it open as a feature request, as it's not currently planned.
The complexity of the implementation for this specific feature means that its unfortunately not something we can whip out in a few days 😅 
I will keep this request updated as we progress.

**Comment by sitereactor** [comment]
Reopened - this feature request is still relevant
